src/System.CommandLine/Option.cs

```diff
@@ -56,7 +56,7 @@ public virtual Argument Argument
 
         private protected override void ChooseNameForUnnamedArgument(Argument argument)
         {
-            argument.Name = Aliases[0].ToLower();
+            argument.Name = Name.ToLower();
```

Comment:
This is a subtle difference here. Assuming an explicit name is not specified, it appears that going to `Name` will switch to using the longest alias rather than the first one. Should we add another test to cover this behavior?

Reply:
I'm not too worried about testing emergent behaviors until we're out of beta. If someone was relying on this it will break them but I suspect the new behavior is more robust in any case.